receiver/splunkhecreceiver/splunkhec_to_metricdata.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecreceiver
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
+)
+
+// SplunkHecToMetricsData converts Splunk HEC metric points to
+// consumerdata.MetricsData. Returning the converted data and the number of
+// dropped time series.
+func SplunkHecToMetricsData(
+	logger *zap.Logger,
+	splunkHecDataPoints *splunk.Metric,
+) (pdata.Metrics, int) {
+
+	// TODO: not optimized at all, basically regenerating everything for each
+	// 	data point.
+	numDroppedTimeSeries := 0
+	md := pdata.NewMetrics()
+	resourceMetrics := pdata.NewResourceMetrics()
+	resourceMetrics.InitEmpty()
+	resourceMetrics.Resource().InitEmpty()
+	metrics := pdata.NewInstrumentationLibraryMetrics()
+	metrics.InitEmpty()
+
+	labelKeys, labelValues := buildLabelKeysAndValues(splunkHecDataPoints.Fields)
+	for i, k := range labelKeys {
+		resourceMetrics.Resource().Attributes().InsertString(k, labelValues[i])
```

Comment:
After reading the spec I'm really not sure what the fundamental distinction between resource attributes and datapoint labels is.  @tigrannajaryan can you explain it so Antoine can verify that this is appropriate to put all of these as a resource attribute?

Reply:
Resource typically describes the location that emits the metric, while labels don't. For example for "cpu.time" metric, the "host.name" should be a Resource attribute, while "state" (idle, user, system) should be a label.

Also aggregation across labels must be meaningful but they don't have to for Resource attribute. So for example aggregating "cpu.time" across different services is likely pointless so "service.name" should be a Resource attribute.

I am not sure how to best apply these rules to received HEC data. Probably things like "host" and "source" should be  a Resource attribute while the fields are labels.

@bogdandrutu any thoughts?